src/ui/molecules/stepper/Stepper.tsx

```diff
@@ -1,174 +1,119 @@
-import type { Reducer } from 'react'
-import React, { useCallback, useMemo, useReducer } from 'react'
-import type { DeepReadonly, UseReducer } from 'superTypes'
-import './stepper.scss'
-import { Button } from '../../atoms/button/Button'
 import classNames from 'classnames'
-import './i18n/index'
-import { useTranslation } from 'hook/useTranslation'
-import type { UseTranslationResponse } from 'hook/useTranslation'
-import { Typography } from 'ui/atoms/typography/Typography'
-import { List as ImmutableList } from 'immutable'
 import type { Breakpoints } from 'hook/useBreakpoint'
 import { useBreakpoint } from 'hook/useBreakpoint'
+import type { UseTranslationResponse } from 'hook/useTranslation'
+import { useTranslation } from 'hook/useTranslation'
+import React, { useMemo } from 'react'
+import type { DeepReadonly } from 'superTypes'
 import { Icon } from 'ui/atoms/icon/Icon'
+import { Typography } from 'ui/atoms/typography/Typography'
+import { Button } from '../../atoms/button/Button'
+import './i18n/index'
+import './stepper.scss'
 
-export type StepStatus = 'disabled' | 'invalid' | 'completed' | 'active' | 'uncompleted'
-
+/**
+ * The index of a step in a step array.
+ */
 export type StepIndex = number
-
-type StepperState = {
-  enabledSteps: ImmutableList<StepIndex>
-  stepsStatuses: ImmutableList<StepStatus>
-  activeStepIndex: StepIndex
-}
-
-type StepperAction =
-  | { type: 'previousClicked' }
-  | { type: 'stepCompleted' }
-  | { type: 'stepFailed' }
-  | { type: 'stepperSubmitted' }
-  | { type: 'stepperReseted'; payload: ImmutableList<Step> }
+/**
+ * The ID of the step.
+ */
+export type StepId = string
+/**
+ * The status of a step in the Stepper.
+ */
+export type StepStatus = 'disabled' | 'invalid' | 'completed' | 'uncompleted'
+type StepState = StepStatus | 'active'
```

Comment:
Be careful that it's not really a state you want to express here, but more an extended status. (the state is handled by the hook)